community/kernel/src/main/java/org/neo4j/kernel/impl/util/ValueUtils.java

```diff
@@ -326,11 +326,25 @@ else if ( map.containsKey( "latitude" ) && map.containsKey( "longitude" ) )
 
     public static NodeValue fromNodeProxy( Node node )
     {
-        return new NodeProxyWrappingNodeValue( node );
+        if ( node instanceof NodeValue )
+        {
+            return (NodeValue) node;
+        }
+        else
+        {
+            return new CoreNodeWrappingNodeValue( node );
+        }
     }
 
     public static RelationshipValue fromRelationshipProxy( Relationship relationship )
     {
-        return new RelationshipProxyWrappingValue( relationship );
+        if ( relationship instanceof RelationshipValue )
+        {
+            return (RelationshipValue) relationship;
+        }
+        else
+        {
+            return new CoreRelationshipWrappingValue( relationship );
+        }
```

Comment:
Same thing goes for here as for `fromNodeProxy()` - we can treat `RelationshipProxy` special even if it doesn't extend `RelationshipValue`.